docs/development/developer.rst

```diff
@@ -4,69 +4,108 @@ Developer Guide
+
+All the available algorithms are present in ``algorithms/`` with each algorithm in a distinct file. 
+Adversarial algorithms like AIRL and GAIL are present in ``algorithms/adversarial``.
 
 
 Data
 ----
 .. automodule:: imitation.data
     :noindex:
 
-Envs
-----
-.. automodule:: imitation.envs
-    :noindex:
+    ``data.wrapper.BufferingWrapper``: Wraps a ``VecEnv`` to save the trajectories from all the environments
+    in a buffer.
 
-Policies
---------
-.. automodule:: imitation.policies
-    :noindex:
+    ``data.wrapper.RolloutInfoWrapper``: Wraps a ``gym.Env`` to log the original observations and rewards recieved from 
+    the environment. The observations and rewards of the entire episode is logged in the ``info`` dictionary with the
+    key "rollout" of the step where the episode ends. This wrapper is useful to save rollout trajectories, especially
+    in cases where you want to bypass the reward and/or observation overrides from other wrappers. 
+    See ``data.rollout.unwrap_traj`` for details and ``scripts/train_rl.py`` for example usecase.
+
+    ``data.rollout.rollout``: Generates rollout by taking in any policy as input along with the environment. 
+
+.. Envs
+.. ----
+.. .. automodule:: imitation.envs
+..     :noindex:
+
+.. Policies
+.. --------
+.. .. automodule:: imitation.policies
+..     :noindex:
 
 Rewards
 -------
 .. automodule:: imitation.rewards
     :noindex:
 
+    ``rewards.reward_wrapper.RewardVecEnvWrapper``: This class wraps a ``VecEnv`` with a custom ``RewardFn``. 
+    The default reward function of the environment is overridden with the passed reward function 
+    and the original rewards are stored in the ``info_dict`` with the ``original_env_rew`` key. 
+    This class is used to override the original reward function of an environment with a learned 
+    reward function from the reward learning algorithms like preference comparisons.
 
 Scripts
 -------
-.. automodule:: imitation.scripts
-    :noindex:
+.. .. automodule:: imitation.scripts
+..     :noindex:
+
+We use Sacred to provide a command-line interface to run the experiments. The scripts to run the end-to-end experiments are
+available in ``scripts/``. You can take a look at the following doc links to understand how to use Sacred:
+
+- `Experiment Overview <https://sacred.readthedocs.io/en/stable/experiment.html>`_: Explains how to create and run experiments. All the experiment objects are defined in ``scripts/config`` in the respective algorithm name file and their ``main`` functions are defined in algorithm name file in ``scripts/``. For example, ``train_rl_ex`` object is defined in ``scripts.config.train_rl`` and its main function is in ``scripts.train_rl``.
+
+- `Ingredients <https://sacred.readthedocs.io/en/stable/ingredients.html>`_: Explains how to use ingredients to avoid code duplication across experiments. The ingredients used in our experiments are defined in ``scripts/common/``:
+  
+  .. autosummary:: 
```

Comment:
This is good! One of the most confusing parts of `imitation` for newcomers, will definitely help new devs.